uniswap_universal_router_decoder/_encoder.py

```diff
@@ -73,6 +73,17 @@ class PathKey(TypedDict):
     hook_data: bytes
 
 
+Permit2TransferDetail = TypedDict(
```

Comment:
Let's stick to the name used by Uniswap: `Permit2TransferDetail` -> `AllowanceTransferDetails`

Also, should be made available in `__init__.py`.